odl/solvers/smooth/gradient.py

```diff
@@ -110,6 +110,77 @@ def steepest_descent(f, x, line_search=1.0, maxiter=1000, tol=1e-16,
             callback(x)
 
 
+def adam(f, x, learning_rate=1e-3, beta1=0.9, beta2=0.999, eps=1e-8,
+         maxiter=1000, tol=1e-16, callback=None):
+    """ADAM method to minimize an objective function.
+
+    General implementation of ADAM for solving
+
+    .. math::
+        \min f(x)
+
+    The algorithm is intended for unconstrained problems.
+
+    The algorithm is described in
+    `Adam: A Method for Stochastic Optimization
+    <https://arxiv.org/abs/1412.6980>`_. All parameter names are taken from
```

Comment:
Fair suggestion. I don't insist on a `References` section, but we need some way of handling papers or books where we cannot point to an online resource (or when it's behind a paywall). There I think we should stick to classical citation style. We don't have to make it clickable but make sure that the full citation appears somewhere, best in the `References` section. Like so:
```
Citing [ABC2016] here.

References
---------
[ABC2016] ...
```
Or so, clickable and explicitly printed (but looks weird in the plain doc):
```
Citing [ABC2016]_ here.

References
---------
.. [ABC2016] ...
[ABC2016}
```